application/comit_node/src/connection_pool.rs

```diff
@@ -0,0 +1,286 @@
+use crate::{
+    bam_api::rfc003::swap_config, node_id::NodeId, swap_protocols::rfc003::bob::BobSpawner,
+};
+use bam::{self, json};
+use futures::{
+    future::{self, Shared},
+    Future,
+};
+use std::{
+    collections::HashMap,
+    fmt::{self, Debug},
+    io,
+    net::SocketAddr,
+    sync::{Arc, Mutex, RwLock},
+};
+use tokio::{
+    self,
+    io::{AsyncRead, AsyncWrite},
+    net::TcpStream,
+};
+
+type BamClient =
+    Arc<Mutex<bam::client::Client<json::Frame, json::OutgoingRequest, json::Response>>>;
+
+enum ConnectionState {
+    Connected {
+        client: BamClient,
+    },
+    Connecting {
+        waiting_list: Shared<Box<dyn Future<Item = BamClient, Error = io::ErrorKind> + Send>>,
+    },
+    Disconnected,
+}
+
+impl Debug for ConnectionState {
```

Comment:
:wrench: Instead of doing this, you could also just [`ignore`](https://mcarton.github.io/rust-derivative/Debug.html#ignoring-a-field) the inner fields of `ConnectionState`:

```rust
#[derive(Derivative)]
#[derivative(Debug)]
enum ConnectionState {
    Connected {
        #[derivative(Debug="ignore")]
        client: BamClient,
    },
    Connecting {
        #[derivative(Debug="ignore")]
        waiting_list: Shared<Box<dyn Future<Item = BamClient, Error = io::ErrorKind> + Send>>,
    },
    Disconnected,
}
```

This would allow you to remove the manual `Debug` and `Display` impl.